lib/stripe/stripe_client.rb

```diff
@@ -17,11 +19,14 @@ class StripeClient
     #
     # Takes a connection manager object for backwards compatibility only, and
     # that use is DEPRECATED.
-    def initialize(_connection_manager = nil)
+    def initialize(_connection_manager = nil, api_key: nil)
```

Comment:
We should probably support all the configuration options that `Stripe` does like `stripe_account`, api_base`, ..., but also even things like `open_timeout`.

This is going to be a more complex refactor though, so I could probably take this job.

Reply:
Happy to take a stab at that unless you'd prefer to keep this PR smaller?

Reply:
Totally up to you :) I don't mind doing it and offered because it's a little bit of work, but if you want to take a stab at it, that'd be great too. I haven't looked too deeply, but it might be a little non-trivial because most of them are checked in pretty random places throughout the codebase.